### PR TITLE
Ensure default serviceId for a CustomerIdentifier is protected

### PR DESCRIPTION
**Pull request by maheshkelkar**
- Use ParamMap instead of req.uri, since it fetches query param
  from uri (GET) or contentString (POST)
- Log an error message if we see multiple ServiceIdentifiers
  for a tuple of (service name and protected flag.

### CODE REVIEW

**Comment by maheshkelkar** [comment]
@pfc please hold off this review. I will let you know when its ready

**Comment by maheshkelkar** [comment]
Abandoning, will submit a new one

File: README.md [hunk 1]
```diff
@@ -44,7 +44,7 @@ libraryDependencies ++= Seq(
 
 ```scala
 libraryDependencies ++= Seq(
-  "com.lookout.borderpatrol" %% "[borderpatrol-module]" % "0.1.18-SNAPSHOT"
+  "com.lookout.borderpatrol" %% "[borderpatrol-module]" % "0.1.20-SNAPSHOT"
```
Comment (PeterFCaswell) [inline README.md]: What happened to 0.1.19?

File: auth/src/main/scala/com/lookout/borderpatrol/auth/keymaster/Keymaster.scala [hunk 2]
```diff
@@ -91,14 +91,15 @@ object Keymaster {
   case class KeymasterTransformFilter(oAuth2CodeVerify: OAuth2CodeVerify)(implicit statsReceiver: StatsReceiver)
       extends Filter[BorderRequest, Response, KeymasterIdentifyReq, Response] {
 
-    def transformInternal(req: BorderRequest): Future[InternalAuthCredential] =
+    def transformInternal(req: BorderRequest): Future[InternalAuthCredential] = {
       (for {
-        u <- Helpers.scrubQueryParams(req.req.uri, "username")
-        p <- Helpers.scrubQueryParams(req.req.uri, "password")
+        u <- Helpers.scrubQueryParams(req.req.params, "username")
```
Comment (PeterFCaswell) [inline auth/src/main/scala/com/lookout/borderpatrol/auth/keymaster/Keymaster.scala]: Not something to delay this PR for, but would it be cleaner if you defined a method in BorderRequest, called "params", that simply invoked "req.params"?

File: core/src/main/scala/com/lookout/borderpatrol/util/Helpers.scala [hunk 3]
```diff
@@ -1,35 +1,20 @@
...
-import org.jboss.netty.handler.codec.http.QueryStringDecoder
+import com.twitter.finagle.http.ParamMap
 
-import scala.collection.JavaConverters._
-import scala.util.{Success, Try}
 
 object Helpers {
 
-  /** Parse uri into (path, params) */
+  /** Regular Expression for all special characters from ASCII 0 to 20 */
```
Comment (PeterFCaswell) [inline core/src/main/scala/com/lookout/borderpatrol/util/Helpers.scala]: Is this Octal 0-20 or decimal 0-20? If the latter, what about the characters from NAK to SP?

**Pull request closed without merging**